### PR TITLE
misc improvements to program exit handling

### PR DESCRIPTION
**Pull request by fabiangreffrath**
…e exit sequence

### CODE REVIEW

File: Source/i_system.c [hunk 1]
```diff
@@ -285,6 +286,7 @@ void I_Init(void)
   // killough 3/6/98: end of keyboard / autorun state changes
 
    I_AtExit(I_Shutdown, true);
+   I_AtExitPrio(I_ErrorMsg, true, "I_ErrorMsg", exit_priority_verylast);
```
Comment (fabiangreffrath) [inline Source/i_system.c:289]: Should I provide a "very very last" category, or simply apply sequence numbers, or just keep it? Currently, `Z_Close()` is executed one step later than this.

**Comment by rfomin** [comment]
Now I've got this if press Ctrl-C:
```
I_SignalHandler: Exit on Signal 2
Exit Sequence[0]: I_QuitFirst (-1)
Exit Sequence[1]: I_ShutdownGraphics (-1)
Exit Sequence[1]: D_QuitNetGame (-1)
Exit Sequence[1]: active_module->I_ShutdownMusic (-1)
Exit Sequence[1]: active_module->I_ShutdownMusic (-1)
Exit Sequence[1]: I_ShutdownSound (-1)
Exit Sequence[1]: I_Shutdown (-1)
Exit Sequence[2]: I_Quit (-1)
I_SignalHandler: Exit on Signal 11
Exit Sequence[3]: I_ErrorMsg (-1)
Exit Sequence[3]: Z_Close (-1)
```
This is normal?

**Comment by rfomin** [comment]
I fixed Signal 11 exit with this change:
```diff
--- a/Source/i_video.c
+++ b/Source/i_video.c
@@ -1597,6 +1597,7 @@ void I_QuitVideo (int phase)
     }
     if (sdlscreen != NULL)
     {
+      sdlscreen->pixels = NULL;
       SDL_FreeSurface(sdlscreen);
       sdlscreen = NULL;
     }
```

File: Source/i_system.c (hunk 1 above)
Comment (rfomin) [reply Source/i_system.c:289]: I think it's ok to keep it as is. The exit sequence already seems complicated.

**Comment by fabiangreffrath** [comment]
> I fixed Signal 11 exit with this change:

Yes, I suspected that it had to do with this. However, if you set it to `NULL`, then SDL has no chance to free the frame buffer. Does it help if you do `screens[0] = NULL` instead? It is a pointer to `sdlscreen->pixels`.

I don't know how this could still get dereferenced somewhere at this point in the code and why pointing it to `NULL` should help with this.

**Comment by rfomin** [comment]
> Does it help if you do `screens[0] = NULL` instead?

No, still Signal 11.

I suggest removing `I_QuitVideo()`, Choco/Crispy don't do that, I don't think it's necessary.

**Review by rfomin — APPROVED** [review]

**Pull request merged**